README.md

```diff
@@ -45,14 +45,47 @@ const formattedCreditCardInput = new RestrictedInput({
 });
 ```
 
+## Patterns
+
+Patterns are a mixture of [`Placeholder`](#placeholder)s and [`PermaChar`](#permachar)s.
+
+### Placeholder
+
+A `Placeholder` is the part of the pattern that accepts user input based on some restrictions. A placeholder is defined in the pattern using two open curly brackets, the placeholder, followed by two closing curly brackets e.g. `{{Abc123}}`.
+
+The patterns a `Placeholder` can be are:
+ - a single alpha character that matches the alpha regex `[A-Za-z]`. e.g. `{{C}}` will match one alpha character.
+ - a single digit that matches the digit regex `\d`. e.g. `{{3}}` will match one digit.
+ - a wildcard character that matches `/./`. e.g. `{{$}}` will match the next character.
```

Comment:
Ah I misread. The wildcard character in patterns is only `*`.

I'll fix this, and add some integration tests to show this behavior.